### PR TITLE
Fix -fsanitize=address bugs

### PR DESCRIPTION
**Pull request by jedbrown**
note: ‘snprintf’ output between 21 and 31 bytes into a destination of size 30

The assembly test had inconsistent ordering of arguments between the qfunction itself and the call to CeedQFunctionAddInput.

### CODE REVIEW

File: tests/t568-operator.c [hunk 1]
```diff
@@ -72,8 +72,8 @@ int main(int argc, char **argv) {
   CeedQFunctionAddOutput(qf_setup, "qdata", dim * (dim + 1) / 2, CEED_EVAL_NONE);
 
   CeedQFunctionCreateInterior(ceed, 1, diff, diff_loc, &qf_diff);
-  CeedQFunctionAddInput(qf_diff, "qdata", dim * (dim + 1) / 2, CEED_EVAL_NONE);
   CeedQFunctionAddInput(qf_diff, "du", num_comp * dim, CEED_EVAL_GRAD);
+  CeedQFunctionAddInput(qf_diff, "qdata", dim * (dim + 1) / 2, CEED_EVAL_NONE);
   CeedQFunctionAddInput(qf_diff, "dummy u", num_comp, CEED_EVAL_INTERP);
```
Comment (jedbrown) [inline tests/t568-operator.c:77]: Sadly our test didn't check for correctness, just that it executed. We should activate `make ASAN=1` (i.e., `-fsanitize=address`) in one of our tests so stuff like this doesn't happen unnoticed.
```
==1842719==ERROR: AddressSanitizer: heap-buffer-overflow on address 0x611000000ed8 at pc 0x55d0d242482b bp 0x7ffcaa7cfd90 sp 0x7ffcaa7cfd80
READ of size 8 at 0x611000000ed8 thread T0
    #0 0x55d0d242482a in diff /home/jed/src/libCEED/tests/t568-operator.h:51
    #1 0x7f0c424b2763 in CeedQFunctionApply_Ref /home/jed/src/libCEED/backends/ref/ceed-ref-qfunction.c:37
    #2 0x7f0c4249104e in CeedQFunctionApply /home/jed/src/libCEED/interface/ceed-qfunction.c:966
    #3 0x7f0c424b0cac in CeedOperatorLinearAssembleQFunctionCore_Ref /home/jed/src/libCEED/backends/ref/ceed-ref-operator.c:483
    #4 0x7f0c424b1239 in CeedOperatorLinearAssembleQFunction_Ref /home/jed/src/libCEED/backends/ref/ceed-ref-operator.c:510
    #5 0x7f0c42482197 in CeedOperatorLinearAssembleQFunctionBuildOrUpdate /home/jed/src/libCEED/interface/ceed-preconditioning.c:1558
    #6 0x7f0c424757ab in CeedSingleOperatorAssemble /home/jed/src/libCEED/interface/ceed-preconditioning.c:570
    #7 0x7f0c42485250 in CeedOperatorLinearAssemble /home/jed/src/libCEED/interface/ceed-preconditioning.c:1959
    #8 0x7f0c42484f1a in CeedOperatorLinearAssemble /home/jed/src/libCEED/interface/ceed-preconditioning.c:1942
    #9 0x55d0d2426277 in main /home/jed/src/libCEED/tests/t568-operator.c:107
    #10 0x7f0c4223984f  (/usr/lib/libc.so.6+0x2384f) (BuildId: 2f005a79cd1a8e385972f5a102f16adba414d75e)
    #11 0x7f0c42239909 in __libc_start_main (/usr/lib/libc.so.6+0x23909) (BuildId: 2f005a79cd1a8e385972f5a102f16adba414d75e)
    #12 0x55d0d24242c4 in _start (/home/jed/src/libCEED/build/t568-operator+0x22c4) (BuildId: cafa255a93b23dec0962ce1748fe85cd5dbc4b68)

0x611000000ed8 is located 0 bytes after 216-byte region [0x611000000e00,0x611000000ed8)
allocated by thread T0 here:
    #0 0x7f0c42ce0cd1 in __interceptor_calloc /usr/src/debug/gcc/gcc/libsanitizer/asan/asan_malloc_linux.cpp:77
    #1 0x7f0c42444660 in CeedCallocArray /home/jed/src/libCEED/interface/ceed.c:231
    #2 0x7f0c424baee1 in CeedVectorSetArray_Ref /home/jed/src/libCEED/backends/ref/ceed-ref-vector.c:66
    #3 0x7f0c42497a0f in CeedVectorSetArray /home/jed/src/libCEED/interface/ceed-vector.c:253
    #4 0x7f0c424bba4c in CeedVectorGetArrayWrite_Ref /home/jed/src/libCEED/backends/ref/ceed-ref-vector.c:144
    #5 0x7f0c424993a7 in CeedVectorGetArrayWrite /home/jed/src/libCEED/interface/ceed-vector.c:430
    #6 0x7f0c424b4dfa in CeedElemRestrictionApply_Ref_Core /home/jed/src/libCEED/backends/ref/ceed-ref-restriction.c:38
    #7 0x7f0c424b6f8d in CeedElemRestrictionApply_Ref_310 /home/jed/src/libCEED/backends/ref/ceed-ref-restriction.c:188
    #8 0x7f0c424b75fe in CeedElemRestrictionApply_Ref /home/jed/src/libCEED/backends/ref/ceed-ref-restriction.c:249
    #9 0x7f0c4244de63 in CeedElemRestrictionApply /home/jed/src/libCEED/interface/ceed-elemrestriction.c:701
    #10 0x7f0c424aba07 in CeedOperatorSetupInputs_Ref /home/jed/src/libCEED/backends/ref/ceed-ref-operator.c:165
    #11 0x7f0c424af4ce in CeedOperatorLinearAssembleQFunctionCore_Ref /home/jed/src/libCEED/backends/ref/ceed-ref-operator.c:402
    #12 0x7f0c424b1239 in CeedOperatorLinearAssembleQFunction_Ref /home/jed/src/libCEED/backends/ref/ceed-ref-operator.c:510
    #13 0x7f0c42482197 in CeedOperatorLinearAssembleQFunctionBuildOrUpdate /home/jed/src/libCEED/interface/ceed-preconditioning.c:1558
    #14 0x7f0c424757ab in CeedSingleOperatorAssemble /home/jed/src/libCEED/interface/ceed-preconditioning.c:570
    #15 0x7f0c42485250 in CeedOperatorLinearAssemble /home/jed/src/libCEED/interface/ceed-preconditioning.c:1959
    #16 0x7f0c42484f1a in CeedOperatorLinearAssemble /home/jed/src/libCEED/interface/ceed-preconditioning.c:1942
    #17 0x55d0d2426277 in main /home/jed/src/libCEED/tests/t568-operator.c:107
    #18 0x7f0c4223984f  (/usr/lib/libc.so.6+0x2384f) (BuildId: 2f005a79cd1a8e385972f5a102f16adba414d75e)
```

**Review by jeremylt — APPROVED** [review]
LGTM. Does this fix all our remaining ASAN issues?

**Comment by jedbrown** [comment]
I'm not sure. What other issues are you aware of? I got a clean run of the standard tests.

**Comment by jeremylt** [comment]
I think this was the last lingering one I knew about.

**Review by jeremylt — APPROVED** [review]

**Pull request merged**